Rock, Paper, Scissors, Lizard, Spock

Caitlin-cooling opened the pull request.

Caitlin-cooling committed:
Functionality to enter namefrom the home page, then be shown a greeting. User then has 3 buttons to click that are the moves they can play. There is not a game class that looks after the random generation of the games move

Caitlin-cooling committed:
Selecting an option  now redirects you to a confirmation page

Caitlin-cooling committed:
Moved win_or_lose file into view to resolve bug

Caitlin-cooling committed:
Add winner class

Caitlin-cooling committed:
Fleshed out the winner tests to test ever scenario

Caitlin-cooling committed:
Update README

Caitlin-cooling committed:
Fix rubocop issues, and write tests for confirmation page

Caitlin-cooling committed:
select option now stubs sample to test confirmation page

Caitlin-cooling committed:
Using params

Caitlin-cooling committed:
Refactor to make one form with 3 inputs, use params to differentiate

Caitlin-cooling committed:
Fiz some rubocop issues and refactor winner messages

Caitlin-cooling committed:
CSS

Caitlin-cooling committed:
Add try again button

Caitlin-cooling committed:
Add 2 fields to root route, and create turn class

Caitlin-cooling committed:
Player1 move redirects to game2

Caitlin-cooling committed:
Add players class, player1 plays first, then player2

Caitlin-cooling committed:
Functionality to determine the winner out of 2 player

Caitlin-cooling committed:
Update README to reflect 2 players

Caitlin-cooling committed:
Fleshed out winner tests and refeactored win_or_lose

Caitlin-cooling committed:
Refactor to ensure that posts have a single responsibility

Caitlin-cooling committed:
Root now asks how many players you would like. 2 points to current flow, and 1 points to oneplayer pager

Caitlin-cooling committed:
Player one shows that you are playing the computer and confirms what you move was

Caitlin-cooling committed:
Shows you the computers move and who won. single player now stubs randomness

Caitlin-cooling committed:
Update REAMDE

Caitlin-cooling committed:
Refactor

Caitlin-cooling committed:
Rename paths/files to be more readable

Caitlin-cooling committed:
Refactor winner

Caitlin-cooling committed:
Begin change over to include laizrd, spock. Change background colour

Caitlin-cooling committed:
Refactor winner spec to cover new possible outcomes

Caitlin-cooling committed:
Fix multiplayer game2 bug

Caitlin-cooling committed:
Styling fixes

Caitlin-cooling committed:
Convert moves to symbols

Caitlin-cooling committed:
Create messages module that looks after win messages

Caitlin-cooling committed:
Refactor and update README to include issues

Caitlin-cooling committed:
Refactor

Caitlin-cooling committed:
Update README to outline issues

soph-g commented on README.md:
```diff
@@ -1,86 +1,27 @@
-Instructions
--------
+### How do I Use it?
+Clone this repo. Run the ruby file `app.rb`, this will start your server. Then in your browser go to localhost:4567.
```
Try to give exact commands, and don't forget to include installing gems and running tests in the instructions.

Caitlin-cooling replied:
Updated how do I use. Running tests is in a separate section below.

soph-g commented on README.md:
````diff
@@ -1,86 +1,27 @@
-I would like to be able to play rock/paper/scissors
-```
+### Known Issues
+* Line 6 in `winner.rb` is too long. I plan to use a hash instead of an array and access the values of that the see player 2 move
````
These are all good changes to identify. If you can spot opportunities for improvement it's good to actually make those changes before submitting for a coach review.

soph-g commented on app.rb:
```diff
@@ -0,0 +1,84 @@
+    erb(:index)
+  end
+
+  post '/two_player' do
```
Not passing any data here, should this be a post route?

Caitlin-cooling replied:
Changed

soph-g commented on app.rb:
```diff
@@ -0,0 +1,84 @@
+    erb(:two_player)
+  end
+
+  post '/one_player' do
```
Not passing any data, should this be a post route?

soph-g commented on app.rb:
```diff
@@ -0,0 +1,84 @@
+    erb(:multiplayer_game)
+  end
+
+  post '/multiplayer_game2_names' do
```
Not sure what this route does, naming is not clear.

Caitlin-cooling replied:
Have since learnt that you should post to the current page then redirect, have implemented this.

soph-g commented on app.rb:
```diff
@@ -0,0 +1,84 @@
+
+  get '/game/result' do
+    @players = Players.instance
+    @name1 = @players.name1
```
Lots of variables being set here - is there a way to avoid this and use a single variable that the view can access the moves and player names on?

soph-g commented on capybara-201810200910053459868959.html:
```diff
@@ -0,0 +1,6366 @@
+<!DOCTYPE html>
```
Don't think this file should be commited - looks like it's been generated by capybara.

Caitlin-cooling replied:
Deleted

soph-g commented on lib/winner.rb:
```diff
@@ -0,0 +1,32 @@
+class Winner
+  include Messages
+
+  WIN_COMBOS = [[:rock, :scissors], [:paper, :rock], [:scissors, :paper], [:scissors, :lizard], [:rock, :lizard], [:lizard, :paper], [:paper, :spock], [:lizard, :spock], [:spock, :rock], [:spock, :scissors]]
```
Find this structure unclear - what do the nested arrays represent? Difficult to work out what the rules are using this information.

soph-g commented on lib/winner.rb:
```diff
@@ -0,0 +1,32 @@
+  end
+
+  def determine
+    WIN_COMBOS.each do |combo|
```
Could you simplify this to avoid using if statements and iteration?

soph-g commented on lib/winner.rb:
```diff
@@ -0,0 +1,32 @@
+
+  private
+
+  def convert_to_sym
```
Could this method be simplified to remove repetition?

soph-g commented on spec/players_spec.rb:
```diff
@@ -0,0 +1,21 @@
+
+  let(:players) { Players.new("Caitlin", "James") }
+
+  it 'name1 returns the name of the first player' do
```
Recommend grouping unit tests by the method under test using 'describe' blocks.

soph-g commented on lib/players.rb:
```diff
@@ -0,0 +1,18 @@
+class Players
```
Feels like this class has multiple responsibilities. It's holding the names and moves for both player 1 and player 2. Is there a Player object that could be extracted here?

soph-g commented on lib/computer.rb:
```diff
@@ -0,0 +1,9 @@
+  attr_reader :move2, :name2
+
+  def initialize(options)
+    @move2 = options.sample
```
Why move2 and name2?

Caitlin-cooling committed:
Update README

Caitlin-cooling committed:
Change players class [WIP]

- Now just takes one name and saves one move
- Class method that returns all instances of the Players object
- Updated app.rb to reflect

Caitlin-cooling committed:
Players display

- Players names are showing in the correct order
- Added reverse to the Players all class method as the ObjectSpace array 
was in the wrong order

Caitlin-cooling committed:
Multiplayer

- Now working with the new player code
- Tests seem to be failing randomly, will try again with more reliable 
wifi

Caitlin-cooling committed:
Single player

- Now works with Player changes. Have tested mnaually, tests seems to be 
failing randomly, am going to try with more reliable wifi

Caitlin-cooling committed:
Unit tests

- Moved in to units folder
- Updated player doubles in winner to reflect new code

Caitlin-cooling committed:
Win or lose spec

- Updated test to reflect new player code
- Tests are failing randomly, am going to try with more raliable wifi

Caitlin-cooling committed:
Players class is now called Player to better reflect what it is

- Have manually tested all functionality passing, testing are still 
failing randomly

Caitlin-cooling committed:
Computer class

- Inherits from player to get it's attr readers
- Options have been added a a constant so they only need to be changed 
here if the options are changed

Caitlin-cooling committed:
Pass Rubocop

- Changed winning combos to be a hash to look up the combos

Caitlin-cooling committed:
Made routing more clear

Caitlin-cooling committed:
Change options to strings to match the players selections

Caitlin-cooling committed:
Refactor app.rb and views accordingly

Caitlin-cooling committed:
Styling

- Added header images
- Changes routing to allow images to display properly
-

Caitlin-cooling committed:
Update README to reflect improvements

Caitlin-cooling committed:
Removing old comments

Caitlin-cooling committed:
Updated Sinatra 2.0.0 -> 2.0.5

Caitlin-cooling committed:
Updated Capybara newest 2.17.0 -> 3.12.0

Caitlin-cooling committed:
Updates rspec newest  3.7.0 -> 3.8.0

Caitlin-cooling committed:
Game class

- Game class stores the players of the game
- New game is created every time the names are entered

Caitlin-cooling committed:
Remove ObjectSpace from players as this is now handled by Game

- All tests passing 🥳

Caitlin-cooling committed:
Update test to use new game method

- Passing rubocop

Caitlin-cooling committed:
Update winner to remove the need to confusing convert to sym method

Caitlin-cooling committed:
Update README.md

Caitlin-cooling committed:
Updated README

Caitlin-cooling committed:
Update README.md

Caitlin-cooling committed:
Updated Travis to include Rubocop and language

Caitlin-cooling committed:
Merge branch 'master' of https://github.com/Caitlin-cooling/rps-challenge

Caitlin-cooling committed:
Updated README, testing Travis

Caitlin-cooling committed:
Merge pull request #1 from Caitlin-cooling/travis

Updated README, testing Travis

Caitlin-cooling committed:
Update README.md

Caitlin-cooling committed:
Adding prcofile for heroku:

Caitlin-cooling committed:
Merge pull request #2 from Caitlin-cooling/travis

Adding procfile for heroku

Caitlin-cooling committed:
Trying to fix Heroku

Caitlin-cooling committed:
Trying to fix Heroku

Caitlin-cooling committed:
Update README.md
